sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -353,7 +353,18 @@ case class AlterTableChangeColumnCommand(
     val newDataSchema = table.dataSchema.fields.map { field =>
       if (field.name == originColumn.name) {
         // Create a new column from the origin column with the new comment.
-        addComment(field, newColumn.getComment)
+        val withNewComment: StructField =
+          addComment(field, newColumn.getComment)
+        // Create a new column from the origin column with the new current default value.
+        if (newColumn.getCurrentDefaultValue().isDefined) {
+          if (newColumn.getCurrentDefaultValue().get.nonEmpty) {
+            addCurrentDefaultValue(withNewComment, newColumn.getCurrentDefaultValue())
+          } else {
+            withNewComment.clearCurrentDefaultValue()
```

Comment:
What if the column is string type and the default value is set as `""`?

Reply:
Good question. I confirmed that the parser returns the quotes in the default value string, and that passing in a completely empty string is prohibited (since the parser expects a token there). I added a couple unit tests to the `DDLParserSuite` to cover this.